File: Android/GameControllerSimulator/app2/src/main/java/cn/maoyanluo/gamecontrollersimulator2/constant/EntityType.kt
package cn.maoyanluo.gamecontrollersimulator2.constant

import cn.maoyanluo.gamecontrollersimulator2.bean.DeviceInfo
import cn.maoyanluo.gamecontrollersimulator2.bean.FeedbackReceived
import com.google.gson.JsonElement

object EntityType {
    const val TYPE_REQUEST_CLIENT_ID = 1
    const val TYPE_REQUEST_CLIENT_ID_RESULT = 2
    const val TYPE_UNREGISTER_CLIENT_ID = 3
    const val TYPE_UNREGISTER_CLIENT_ID_RESULT = 4
    const val TYPE_SEND_GAME_EVENT = 5
    const val TYPE_SEND_GAME_EVENT_RESULT = 6
    const val TYPE_FEEDBACK_RECEIVED = 7
    const val TYPE_FEEDBACK_RECEIVED_RESULT = 8
    const val TYPE_QUERY_CLIENT_INFO = 9
    const val TYPE_QUERY_CLIENT_INFO_RESULT = 10
    const val TYPE_NEW_TYPE_CONNECT = 11
    const val TYPE_NEW_TYPE_CONNECT_RESULT = 12
    const val TYPE_ECHO = 13
    const val TYPE_ECHO_RESULT = 14
    const val TYPE_RTT = 15
    const val TYPE_RTT_RESULT = 16

    val TYPE_MAPPING = mapOf(
        Pair(TYPE_REQUEST_CLIENT_ID, JsonElement::class.java),
        Pair(TYPE_REQUEST_CLIENT_ID_RESULT, String::class.java),
        Pair(TYPE_UNREGISTER_CLIENT_ID, String::class.java),
        Pair(TYPE_UNREGISTER_CLIENT_ID_RESULT, JsonElement::class.java),
        Pair(TYPE_SEND_GAME_EVENT, String::class.java),
        Pair(TYPE_SEND_GAME_EVENT_RESULT, JsonElement::class.java),
        Pair(TYPE_FEEDBACK_RECEIVED, FeedbackReceived::class.java),
        Pair(TYPE_FEEDBACK_RECEIVED_RESULT, JsonElement::class.java),
        Pair(TYPE_QUERY_CLIENT_INFO, JsonElement::class.java),
        Pair(TYPE_QUERY_CLIENT_INFO_RESULT, DeviceInfo::class.java),
        Pair(TYPE_NEW_TYPE_CONNECT, String::class.java),
        Pair(TYPE_NEW_TYPE_CONNECT_RESULT, String::class.java),
        Pair(TYPE_ECHO, String::class.java),
        Pair(TYPE_ECHO_RESULT, String::class.java),
        Pair(TYPE_RTT, String::class.java),
        Pair(TYPE_RTT_RESULT, String::class.java)
    )

}

File: iOS/GameControllerSimulator/GameControllerSimulator/Constant/EntityType.swift
//
//  EntityType.swift
//  GameControllerSimulator
//
//  Created by 猫眼螺 on 2026/8/2.
//

struct EmptyData: Codable {}

enum EntityType: Int {
    
    case TYPE_REQUEST_CLIENT_ID = 1
    case TYPE_REQUEST_CLIENT_ID_RESULT = 2
    case TYPE_UNREGISTER_CLIENT_ID = 3
    case TYPE_UNREGISTER_CLIENT_ID_RESULT = 4
    case TYPE_SEND_GAME_EVENT = 5
    case TYPE_SEND_GAME_EVENT_RESULT = 6
    case TYPE_FEEDBACK_RECEIVED = 7
    case TYPE_FEEDBACK_RECEIVED_RESULT = 8
    case TYPE_QUERY_CLIENT_INFO = 9
    case TYPE_QUERY_CLIENT_INFO_RESULT = 10
    case TYPE_NEW_TYPE_CONNECT = 11
    case TYPE_NEW_TYPE_CONNECT_RESULT = 12
    case TYPE_ECHO = 13
    case TYPE_ECHO_RESULT = 14
    case TYPE_RTT = 15
    case TYPE_RTT_RESULT = 16
    
    public static let mapping: [Int: any Decodable.Type] = [
        TYPE_REQUEST_CLIENT_ID.rawValue:
            BaseEntity<EmptyData>.self,
        
        TYPE_REQUEST_CLIENT_ID_RESULT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_UNREGISTER_CLIENT_ID.rawValue:
            BaseEntity<String>.self,
        
        TYPE_UNREGISTER_CLIENT_ID_RESULT.rawValue:
            BaseEntity<EmptyData>.self,
        
        TYPE_SEND_GAME_EVENT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_SEND_GAME_EVENT_RESULT.rawValue:
            BaseEntity<EmptyData>.self,
        
        TYPE_FEEDBACK_RECEIVED.rawValue:
            BaseEntity<FeedbackReceived>.self,
        
        TYPE_FEEDBACK_RECEIVED_RESULT.rawValue:
            BaseEntity<EmptyData>.self,
        
        TYPE_QUERY_CLIENT_INFO.rawValue:
            BaseEntity<EmptyData>.self,
        
        TYPE_QUERY_CLIENT_INFO_RESULT.rawValue:
            BaseEntity<DeviceInfo>.self,
        
        TYPE_NEW_TYPE_CONNECT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_NEW_TYPE_CONNECT_RESULT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_ECHO.rawValue:
            BaseEntity<String>.self,
        
        TYPE_ECHO_RESULT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_RTT.rawValue:
            BaseEntity<String>.self,
        
        TYPE_RTT_RESULT.rawValue:
            BaseEntity<String>.self
    ]
    
}
